app/controllers/api/v1/companies_controller.rb

```diff
@@ -0,0 +1,7 @@
+class Api::V1::CompaniesController < ApplicationController
+
+  def show
+    @company = Company.find(params[:id])
+    render json: @company
+  end
+end
```

Comment:
I am sure other people in this group have the experience to better answer your question about the separate CompaniesController :) Here are my 2 cents: I think it is a very good way to structure the api as this is an endpoint for companies.